src/agent0/chainsync/db/hyperdrive/chain_to_db.py

```diff
@@ -100,3 +113,325 @@ def data_chain_to_db(interface: HyperdriveReadInterface, block: BlockData, sessi
 
     block_pool_info = convert_pool_info(pool_info_dict)
     add_pool_infos([block_pool_info], session)
+
+
+def _event_data_to_dict(in_val: EventData) -> dict[str, Any]:
+    out = dict(in_val)
+    # The args field is also an attribute dict, change to dict
+    out["args"] = dict(in_val["args"])
+
+    # Convert transaction hash to string
+    out["transactionHash"] = in_val["transactionHash"].to_0x_hex()
+    # Get token id field from args.
+    # This field is `assetId` for open/close long/short
+    return out
+
+
+# TODO cleanup
+# pylint: disable=too-many-branches
+# pylint: disable=too-many-statements
+# pylint: disable=too-many-locals
+def trade_events_to_db(
+    interfaces: list[HyperdriveReadInterface],
+    wallet_addr: str,
+    db_session: Session,
+) -> None:
+    """Function to query trade events from all pools and add them to the db.
+
+    Arguments
+    ---------
+    interfaces: list[HyperdriveReadInterface]
+        The hyperdrive interface objects connected to a pool.
```

Comment:
It didn't click at first that we associate an interface with a specific pool. Makes sense now, but maybe updating this docstring to emphasize that **each** object is connected to a pool would help people get it more quickly:

```
A collection of Hyperdrive interface objects, each connected to a pool.
```